### PR TITLE
feat: cwc stacking tearsheets

### PR DESCRIPTION
**Pull request by lee-chase**
### Related Ticket(s)

Builds on #11545

### Description

Adds the stacking feature to tearsheets. It also updates the version of `@carbon/ibm-products-styles`

### Changelog

**Changed**

- packages/carbon-web-components/package.json
- packages/carbon-web-components/src/components/tearsheet/story-styles.scss
- packages/carbon-web-components/src/components/tearsheet/tearsheet-story.ts
- packages/carbon-web-components/src/components/tearsheet/tearsheet.scss
- packages/carbon-web-components/src/components/tearsheet/tearsheet.ts
- yarn.lock

### CODE REVIEW

**Review by ariellalgilmore — COMMENTED** [review]
looks like need to update the `yarn.lock` file. the ci-checks are throwing errors

File: packages/carbon-web-components/src/components/tearsheet/tearsheet-story.ts [hunk 1]
```diff
@@ -315,7 +315,11 @@ const slugs = {
 const getSlug = (index) => {
   switch (index) {
     case 1:
-      return html`<cds-slug slot="slug" className="slug-container" size="xs">
+      return html`<cds-slug
+        slot="slug"
```
[the remark this answers is not in the document]
Comment (lee-chase) [reply packages/carbon-web-components/src/components/tearsheet/tearsheet-story.ts]: Done.

**Review by ariellalgilmore — APPROVED** [review]
LGTM!

**Comment by annawen1** [comment]
Hey there! This issue/pull request was referenced in recently released [v2.5.0](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/releases/tag/v2.5.0).

**Pull request merged**